### PR TITLE
Update index.d.ts Update task interface for specifying dependencies

### PR DESCRIPTION
**Pull request by lifeiscontent**
Add task constructor for dependency specifications

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by lifeiscontent** [comment]
@GiedriusGrabauskas or @tkqubo can either of you review this? I'm unsure how to fix this error on travis.

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

File: types/undertaker/index.d.ts
Comment (GiedriusGrabauskas) [inline types/undertaker/index.d.ts]: Why you removed js-docs comments?

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

File: types/undertaker/index.d.ts
Comment (GiedriusGrabauskas) [inline types/undertaker/index.d.ts]: In source file I see only two arguments:
https://github.com/gulpjs/undertaker/blob/master/lib/task.js#L3
Can you give a link to source or documentation?

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
@lifeiscontent Please address comments from the code reviewers.

**Comment by GiedriusGrabauskas** [comment]
@lifeiscontent, you are [trying to fix](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/20734/files#diff-8d3781da3ed98fb7cd8004e135cc7239R66) `gulp@4` types to be compatible with `gulp@3` ([source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/gulp/v3/index.d.ts#L37)).
If you want to use types for `gulp@3`, you should use this command:
```sh
npm i @types/gulp@3
```
Other changes are only code formatting, which doesn't make sense.

@RyanCavanaugh, I think this pull request should be closed.

File: types/undertaker/index.d.ts
Comment (lifeiscontent) [reply types/undertaker/index.d.ts]: @GiedriusGrabauskas  because I was getting an error saying that this isn't necessary in typescript as its inferred from the implementation.

File: types/undertaker/index.d.ts
Comment (lifeiscontent) [reply types/undertaker/index.d.ts]: @GiedriusGrabauskas https://github.com/gulpjs/gulp/blob/de1acf66a6253302c856971f8c00f65ea6c63882/docs/API.md#gulptaskname--deps--fn

**Comment by lifeiscontent** [comment]
@GiedriusGrabauskas ah, I see. Sorry about that. Haven't used gulp in awhile. Thanks!

**Pull request closed without merging**
